test/hook.lua

```diff
@@ -1,6 +1,19 @@
+buf = ''
+
 -- recv hook, m is message
-function recv(m)
-    print(m)
+function recv(path, m)
+    local code = 0
+    local file = ''
+    buf = buf..m
+    if (string.sub(buf, -4) == '\r\n\r\n')
+    then
+        local method = string.match(buf, "%S+")
+        local name = string.match(buf, "%S+", string.len(method) + 2)
+        code = 200
+        file = path..name
```

Comment:
You got the idea wrong here.
The correct idea is something like this:
```Lua
-- files that are available in the www root
-- use a set so setsockopt can manipulate this table by running `luaL_dostring()`
files = {}
files['foo.txt'] = true
files['boo.txt'] = true

function recv(msg)
   -- ...
   if method == "GET" and file[name] then
       return 200 -- the file is available
   end
end
```

Please adapt the PR accordingly